src/web/mjs/connectors/Dm5.mjs

```diff
@@ -0,0 +1,125 @@
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.title.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        // https://regex101.com/r/vD6dIl/2/
+        const regex = /（(.*?)）/gim;
+        const script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        let button = document.querySelector('#checkAdult');
+                        if(button) {
+                            button.click();
+                        }
+                        let chapterList = [...document.querySelectorAll('${this.queryChapters}')].map(element => {
+                            return {
+                                id: new URL(element.href, window.location).pathname,
+                                title: element.text.replace(${regex}, '').trim()
+                            };
+                        });
+                        resolve(chapterList);
+                    } catch(error) {
+                        reject(error);
+                    }
+                }, 500);
+            });
+        `;
+        const uri = new URL(manga.id, this.url);
+        const request = new Request(uri, this.requestOptions);
+        return await Engine.Request.fetchUI(request, script);
+    }
+
+    async _getPages(chapter) {
+        const script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        let lastPage = 1;
+                        lastPage = parseInt([...document.querySelectorAll('div#chapterpager.chapterpager a')].pop().text.trim());
+                        let pageList = [];
+                        let ajaxResult = null;
+    
+                        for(let dmpage = 1, run = true; dmpage <= lastPage && run; dmpage++){                            
+                            if (ajaxResult != null) {
+                                ajaxResult.abort();
+                                ajaxResult = null;
+                            }
+
+                            let mkey = '';
+                            let data;
+                            if ($("#dm5_key").length > 0) {
+                                mkey = $("#dm5_key").val();
+                            }
+                            ajaxResult = $.ajax({
+                                url: 'chapterfun.ashx',
+                                async: false,
+                                data: { cid: DM5_CID, page: dmpage, key: mkey, language: 1, gtk: 6, _cid: DM5_CID, _mid: DM5_MID, _dt: DM5_VIEWSIGN_DT, _sign: DM5_VIEWSIGN },
+                                type: 'GET',
+                                error: function (msg) {
```

Comment:
This may lead to missing images without the user noticing it, may reject the promise in case an image could not be extracted